### PR TITLE
Blog app - controllers specs

### PR DESCRIPTION
**Pull request by Fombi-Favour**
In this activity, we:
- created a `request` folder in the `spec` directory
- added test for the `users` and `posts` controllers

### CODE REVIEW

File: spec/requests/posts_request_spec.rb [hunk 1]
```diff
@@ -0,0 +1,19 @@
...
+      expect(response.body).to include('All posts list')
+    end
+
+    it 'renders a successful response for a single post' do
+      get '/users/1/posts/1'
+      expect(response).to have_http_status(:success)
+      expect(response.status).to eq(200)
+      expect(response.body).to include('Individual post')
+    end
+  end
```
Comment (DammyShittu) [inline spec/requests/posts_request_spec.rb]: - [x] Kindly separate these two tests into different describe blocks as they test for different controller actions.

  ```
  describe 'GET /index' do
    # index page tests
  end

  describe 'GET /show' do
    # show page tests
  end
  ```

File: spec/requests/users_request_spec.rb [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+      expect(response.body).to include('All users list')
+    end
+
+    it 'renders a successful response for a single user' do
+      get '/users/1'
+      expect(response).to have_http_status(:success)
+      expect(response.status).to eq(200)
+      expect(response.body).to include('Individual user')
+    end
+  end
```
Comment (DammyShittu) [inline spec/requests/users_request_spec.rb]: - [x] Kindly separate these two tests into different describe blocks as they test for different controller actions.

  ```
  describe 'GET /index' do
    # index page tests
  end

  describe 'GET /show' do
    # show page tests
  end
  ```

File: spec/requests/posts_request_spec.rb [hunk 3]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
+  describe 'GET users/:id/posts' do
+    it 'renders a successful response' do
+      get '/users/1/posts'
+      expect(response).to be_successful
+      expect(response.status).to eq(200)
+      expect(response.body).to include('All posts list')
+    end
```
Comment (DammyShittu) [inline spec/requests/posts_request_spec.rb]: - [ ] Kindly add test to assert the index template is rendered.

File: spec/requests/posts_request_spec.rb [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+    it 'renders a successful response for a single post' do
+      get '/users/1/posts/1'
+      expect(response).to have_http_status(:success)
+      expect(response.status).to eq(200)
+      expect(response.body).to include('Individual post')
```
Comment (DammyShittu) [inline spec/requests/posts_request_spec.rb]: - [ ] Kindly add test to assert the show template is rendered.

File: spec/requests/users_request_spec.rb [hunk 5]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe 'Users', type: :request do
+  describe 'GET /users' do
+    it 'renders a successful response' do
+      get '/users'
+      expect(response).to be_successful
+      expect(response.status).to eq(200)
+      expect(response.body).to include('All users list')
+    end
```
Comment (DammyShittu) [inline spec/requests/users_request_spec.rb]: - [ ] Kindly add test to assert the index template is rendered.

File: spec/requests/users_request_spec.rb [hunk 6]
```diff
@@ -0,0 +1,19 @@
...
+
+    it 'renders a successful response for a single user' do
+      get '/users/1'
+      expect(response).to have_http_status(:success)
+      expect(response.status).to eq(200)
+      expect(response.body).to include('Individual user')
+    end
```
Comment (DammyShittu) [inline spec/requests/users_request_spec.rb]: - [ ] Kindly add test to assert the show template is rendered.

File: spec/requests/posts_request_spec.rb [hunk 7]
```diff
@@ -0,0 +1,19 @@
+require 'rails_helper'
+
+RSpec.describe 'Posts', type: :request do
+  describe 'GET users/:id/posts' do
+    it 'renders a successful response' do
```
Comment (DammyShittu) [inline spec/requests/posts_request_spec.rb]: - [ ] Your tests are failing. Kindly fix them.

  image

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Fombi-Favour,

## Highlights
✔️ Rspec test suite

Good job so far!
There are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: spec/requests/posts_request_spec.rb (hunk 3 above)
Comment (Fombi-Favour) [reply spec/requests/posts_request_spec.rb]: Please I tried to add it into the test but it was not working. Can you please tell an alternative way of rendering the template.
I used this: `expect(response).to render_template('users/index')` but not working. Same as for the posts test

File: spec/requests/posts_request_spec.rb (hunk 1 above)
Comment (rloterh) [reply spec/requests/posts_request_spec.rb]: well resolved! ✔️

File: spec/requests/users_request_spec.rb (hunk 2 above)
Comment (rloterh) [reply spec/requests/users_request_spec.rb]: well resolved! ✔️

File: spec/requests/posts_request_spec.rb (hunk 3 above)
Comment (rloterh) [reply spec/requests/posts_request_spec.rb]: well resolved! ✔️

File: spec/requests/posts_request_spec.rb (hunk 4 above)
Comment (rloterh) [reply spec/requests/posts_request_spec.rb]: well resolved! ✔️

File: spec/requests/posts_request_spec.rb (hunk 7 above)
Comment (rloterh) [reply spec/requests/posts_request_spec.rb]: well resolved! ✔️

File: spec/requests/users_request_spec.rb (hunk 5 above)
Comment (rloterh) [reply spec/requests/users_request_spec.rb]: well resolved! ✔️

File: spec/requests/users_request_spec.rb (hunk 6 above)
Comment (rloterh) [reply spec/requests/users_request_spec.rb]: well resolved! ✔️

**Review by rloterh — APPROVED** [review]
## Status: Approved :heavy_check_mark:

Hi Favour,  

Your project is complete! 🏁 There is nothing else to say other than... it's time to merge it 🍾 Congratulations!  🎉💯🌟 

## Highlights 🎯 
🎲 Good job on implementing recommended changes!

## Optional suggestions 🧰 
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you consider them as they can improve your code._

**You can also consider:**
Let's make the ReadMe stand out even better by including instructions under the _Getting Started_ section, on how to run test cases for the app, for instance, using the command `rspec`.

Cheers..Happy coding!..and keep soaring higher! 💻 🍷 🚀 

Feel free to leave any questions or comments in the PR thread if something is not 100% clear. Remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**